### PR TITLE
checker function for converting patients from Record to CQM::Patient

### PR DESCRIPTION
**Pull request by zlister**
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

- [x] cqm-converter branch in Gemfile updated once https://github.com/projecttacoma/cqm-converter/pull/45 merged

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1964
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] ~Tests are included and test edge cases~ (N/A as this is a one time conversion check)
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)

**Reviewer 1:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Comment by lukoktonos** [comment]
Codecov report is buggy--actually it is +0.07 if you go to the codecov site:
https://codecov.io/gh/projecttacoma/bonnie/compare/efeb9456bae24b9bf6a79c69713c5a12192b03d2...bd965f882b0dbd7384ad9695295b730d9773979c

File: lib/tasks/cql.rake [hunk 1]
```diff
@@ -229,13 +229,50 @@ namespace :bonnie do
...
+          diff = patient_conversion_diff(bonnie_patient, cqm_patient)
+          if diff.empty?
+            print ".".green
+          else
+            puts "\nConversion Difference".yellow
+            patient_user = User.find_by(_id: bonnie_patient[:user_id])
+            puts "Patient #{bonnie_patient.first} #{bonnie_patient.last} with id #{bonnie_patient._id} in account #{patient_user.email}".light_blue
+            diff.each_entry do |element|
+              # Get the data element name and remove commas if there are any
+              element = element.split(':')[0].delete(',')
```
Comment (lukoktonos) [inline lib/tasks/cql.rake]: One of the elements in the whitelist, `Diagnostic Study, Result'`, has a comma, so this removal would never match up with that.  I think we need to remove the comma from the whitelist element in order to hit that one, right?

File: lib/tasks/cql.rake [hunk 2]
```diff
@@ -250,7 +287,41 @@ namespace :bonnie do
...
+      differences.push('measure_ids') if record.measure_ids.compact != cqm_patient.measure_ids
+      differences.push('user_id') if record.user_id != cqm_patient.user_id
+      differences.push('expectedValues') if record.expected_values != cqm_patient.expectedValues
+      differences.push('source_data_criteria') if record.source_data_criteria != cqm_patient.qdmPatient.extendedData['source_data_criteria']
+      differences.push('birthDatetime') if Time.at(record.birthdate).utc != cqm_patient.qdmPatient.birthDatetime
+
+      # Retrieve the codeListIds from the dataElements
+      data_elements_code_list_ids = cqm_patient.qdmPatient.dataElements.map(&:codeListId)
+      record.source_data_criteria.each do |sdc|
+        next unless sdc[:id] != 'MeasurePeriod'
```
Comment (lukoktonos) [inline lib/tasks/cql.rake]: I think `next unless ... !=` would be clearer as `next if ... ==`

**Review by lukoktonos — CHANGES_REQUESTED** [review]
In addition to comments here, discussed on slack about changing error printout for `dataElements` and posed the question about there being unexpected differences in our super tester accounts.

File: lib/tasks/cql.rake (hunk 1 above)
Comment (zlister) [reply lib/tasks/cql.rake]: Yep, good catch.

**Review by lukoktonos — APPROVED** [review]

**Review by jbradl11 — CHANGES_REQUESTED** [review]
Looks good. Just need to update the Gemfile.

**Review by jbradl11 — APPROVED** [review]

**Pull request merged**